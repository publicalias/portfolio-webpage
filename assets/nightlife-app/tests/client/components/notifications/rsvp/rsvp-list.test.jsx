"use strict";

//local imports

const RSVPList = require("../../../../../scripts/client/components/notifications/rsvp/rsvp-list");

const { testWrapper } = require("../../../test-helpers");

//global imports

const { initTestSnapshot } = require("redux/tests/client-tests");
const { reactTests } = require("redux/tests/react-tests");

//utilities

const { testShallow } = testWrapper(RSVPList);

//setup

beforeAll(reactTests.setup);
beforeEach(reactTests.inject(RSVPList));

//rsvp list

describe("rsvp list", () => {

  const testSnapshot = initTestSnapshot(testShallow);

  it("should match snapshot (default)", () => testSnapshot());

  it("should match snapshot (bool)", () => testSnapshot({ venues: { data: [{}] } }));

  it("should match snapshot (list)", () => testSnapshot({ notifications: { rsvps: [{}] } }));

});
