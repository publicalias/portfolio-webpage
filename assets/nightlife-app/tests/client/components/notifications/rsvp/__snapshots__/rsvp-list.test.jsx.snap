// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RSVPList should call useRefresh on update 1`] = `
Array [
  Array [
    [MockFunction],
    0,
    Array [],
    Array [
      "FRIEND_GET_LIST",
      "META_GET_USER",
      "RSVP_ADD",
      "RSVP_DISMISS",
      "RSVP_REMOVE",
    ],
  ],
  Array [
    [MockFunction],
    0,
    Array [],
    Array [
      "FRIEND_GET_LIST",
      "META_GET_USER",
      "RSVP_ADD",
      "RSVP_DISMISS",
      "RSVP_REMOVE",
    ],
  ],
]
`;

exports[`RSVPList should match snapshot (default) 1`] = `
<MetaList
  actions={
    Object {
      "favoriteAdd": [MockFunction],
      "favoriteRemove": [MockFunction],
      "friendAdd": [MockFunction],
      "friendConfirm": [MockFunction],
      "friendDismiss": [MockFunction],
      "friendGetList": [MockFunction],
      "friendRemove": [MockFunction],
      "metaAddErrors": [MockFunction],
      "metaCloseError": [MockFunction],
      "metaGetUser": [MockFunction],
      "metaLogAction": [MockFunction],
      "metaNoOp": [MockFunction],
      "metaSaveAddress": [MockFunction],
      "metaSaveAvatar": [MockFunction],
      "metaSetAddress": [MockFunction],
      "metaSetAvatar": [MockFunction],
      "metaSetLoading": [MockFunction],
      "metaSetReady": [MockFunction],
      "metaSetState": [MockFunction],
      "metaTimeoutError": [MockFunction],
      "metaToggleDelete": [MockFunction],
      "metaToggleSettings": [MockFunction],
      "rsvpAdd": [MockFunction],
      "rsvpDismiss": [MockFunction],
      "rsvpGetList": [MockFunction],
      "rsvpRemove": [MockFunction],
      "userClearState": [MockFunction],
      "userGetItem": [MockFunction],
      "userGetList": [MockFunction],
      "userSetSearch": [MockFunction],
      "userToggleBlock": [MockFunction],
      "userToggleRange": [MockFunction],
      "venueClearForm": [MockFunction],
      "venueClearState": [MockFunction],
      "venueGetItem": [MockFunction],
      "venueGetList": [MockFunction],
      "venueSetMessage": [MockFunction],
      "venueSetOpen": [MockFunction],
      "venueSetPause": [MockFunction],
      "venueSetPhoto": [MockFunction],
      "venueSetSearch": [MockFunction],
      "venueSetStart": [MockFunction],
      "venueSetTime": [MockFunction],
      "venueToggleRange": [MockFunction],
      "venueToggleSort": [MockFunction],
    }
  }
  data={
    Object {
      "account": Object {
        "address": "",
        "avatar": "",
        "delete": false,
        "settings": false,
      },
      "errors": Array [],
      "loading": 0,
      "log": Array [],
      "notifications": Object {
        "friends": Array [],
        "rsvps": Array [],
      },
      "ready": false,
      "user": Object {},
      "users": Object {
        "list": Object {
          "data": Array [],
          "range": false,
          "search": "",
        },
        "page": Object {
          "data": Array [],
        },
      },
      "venues": Object {
        "list": Object {
          "data": Array [],
          "range": false,
          "search": "",
          "sort": false,
        },
        "page": Object {
          "data": Array [],
          "form": Object {
            "message": "",
            "open": false,
            "time": "",
          },
          "photos": Object {
            "pause": false,
            "photo": "",
            "start": false,
          },
        },
      },
    }
  }
  history={
    Object {
      "push": [MockFunction],
    }
  }
  local={
    Object {
      "heading": "RSVPs",
      "refresh": [MockFunction],
      "type": "rsvp",
    }
  }
  location={Object {}}
  match={Object {}}
  staticContext={Object {}}
/>
`;

exports[`RSVPList should match snapshot (list) 1`] = `
<MetaList
  actions={
    Object {
      "favoriteAdd": [MockFunction],
      "favoriteRemove": [MockFunction],
      "friendAdd": [MockFunction],
      "friendConfirm": [MockFunction],
      "friendDismiss": [MockFunction],
      "friendGetList": [MockFunction],
      "friendRemove": [MockFunction],
      "metaAddErrors": [MockFunction],
      "metaCloseError": [MockFunction],
      "metaGetUser": [MockFunction],
      "metaLogAction": [MockFunction],
      "metaNoOp": [MockFunction],
      "metaSaveAddress": [MockFunction],
      "metaSaveAvatar": [MockFunction],
      "metaSetAddress": [MockFunction],
      "metaSetAvatar": [MockFunction],
      "metaSetLoading": [MockFunction],
      "metaSetReady": [MockFunction],
      "metaSetState": [MockFunction],
      "metaTimeoutError": [MockFunction],
      "metaToggleDelete": [MockFunction],
      "metaToggleSettings": [MockFunction],
      "rsvpAdd": [MockFunction],
      "rsvpDismiss": [MockFunction],
      "rsvpGetList": [MockFunction],
      "rsvpRemove": [MockFunction],
      "userClearState": [MockFunction],
      "userGetItem": [MockFunction],
      "userGetList": [MockFunction],
      "userSetSearch": [MockFunction],
      "userToggleBlock": [MockFunction],
      "userToggleRange": [MockFunction],
      "venueClearForm": [MockFunction],
      "venueClearState": [MockFunction],
      "venueGetItem": [MockFunction],
      "venueGetList": [MockFunction],
      "venueSetMessage": [MockFunction],
      "venueSetOpen": [MockFunction],
      "venueSetPause": [MockFunction],
      "venueSetPhoto": [MockFunction],
      "venueSetSearch": [MockFunction],
      "venueSetStart": [MockFunction],
      "venueSetTime": [MockFunction],
      "venueToggleRange": [MockFunction],
      "venueToggleSort": [MockFunction],
    }
  }
  data={
    Object {
      "account": Object {
        "address": "",
        "avatar": "",
        "delete": false,
        "settings": false,
      },
      "errors": Array [],
      "loading": 0,
      "log": Array [],
      "notifications": Object {
        "friends": Array [],
        "rsvps": Array [
          Object {
            "date": 0,
            "hidden": Array [],
            "id": "",
            "message": "",
            "time": "",
            "user": Object {
              "id": "",
              "name": "",
            },
            "venue": Object {
              "id": "",
              "name": "",
            },
          },
        ],
      },
      "ready": false,
      "user": Object {},
      "users": Object {
        "list": Object {
          "data": Array [],
          "range": false,
          "search": "",
        },
        "page": Object {
          "data": Array [],
        },
      },
      "venues": Object {
        "list": Object {
          "data": Array [],
          "range": false,
          "search": "",
          "sort": false,
        },
        "page": Object {
          "data": Array [],
          "form": Object {
            "message": "",
            "open": false,
            "time": "",
          },
          "photos": Object {
            "pause": false,
            "photo": "",
            "start": false,
          },
        },
      },
    }
  }
  history={
    Object {
      "push": [MockFunction],
    }
  }
  local={
    Object {
      "heading": "RSVPs",
      "refresh": [MockFunction],
      "type": "rsvp",
    }
  }
  location={Object {}}
  match={Object {}}
  staticContext={Object {}}
>
  <RSVPItem
    actions={
      Object {
        "favoriteAdd": [MockFunction],
        "favoriteRemove": [MockFunction],
        "friendAdd": [MockFunction],
        "friendConfirm": [MockFunction],
        "friendDismiss": [MockFunction],
        "friendGetList": [MockFunction],
        "friendRemove": [MockFunction],
        "metaAddErrors": [MockFunction],
        "metaCloseError": [MockFunction],
        "metaGetUser": [MockFunction],
        "metaLogAction": [MockFunction],
        "metaNoOp": [MockFunction],
        "metaSaveAddress": [MockFunction],
        "metaSaveAvatar": [MockFunction],
        "metaSetAddress": [MockFunction],
        "metaSetAvatar": [MockFunction],
        "metaSetLoading": [MockFunction],
        "metaSetReady": [MockFunction],
        "metaSetState": [MockFunction],
        "metaTimeoutError": [MockFunction],
        "metaToggleDelete": [MockFunction],
        "metaToggleSettings": [MockFunction],
        "rsvpAdd": [MockFunction],
        "rsvpDismiss": [MockFunction],
        "rsvpGetList": [MockFunction],
        "rsvpRemove": [MockFunction],
        "userClearState": [MockFunction],
        "userGetItem": [MockFunction],
        "userGetList": [MockFunction],
        "userSetSearch": [MockFunction],
        "userToggleBlock": [MockFunction],
        "userToggleRange": [MockFunction],
        "venueClearForm": [MockFunction],
        "venueClearState": [MockFunction],
        "venueGetItem": [MockFunction],
        "venueGetList": [MockFunction],
        "venueSetMessage": [MockFunction],
        "venueSetOpen": [MockFunction],
        "venueSetPause": [MockFunction],
        "venueSetPhoto": [MockFunction],
        "venueSetSearch": [MockFunction],
        "venueSetStart": [MockFunction],
        "venueSetTime": [MockFunction],
        "venueToggleRange": [MockFunction],
        "venueToggleSort": [MockFunction],
      }
    }
    data={
      Object {
        "account": Object {
          "address": "",
          "avatar": "",
          "delete": false,
          "settings": false,
        },
        "errors": Array [],
        "loading": 0,
        "log": Array [],
        "notifications": Object {
          "friends": Array [],
          "rsvps": Array [
            Object {
              "date": 0,
              "hidden": Array [],
              "id": "",
              "message": "",
              "time": "",
              "user": Object {
                "id": "",
                "name": "",
              },
              "venue": Object {
                "id": "",
                "name": "",
              },
            },
          ],
        },
        "ready": false,
        "user": Object {},
        "users": Object {
          "list": Object {
            "data": Array [],
            "range": false,
            "search": "",
          },
          "page": Object {
            "data": Array [],
          },
        },
        "venues": Object {
          "list": Object {
            "data": Array [],
            "range": false,
            "search": "",
            "sort": false,
          },
          "page": Object {
            "data": Array [],
            "form": Object {
              "message": "",
              "open": false,
              "time": "",
            },
            "photos": Object {
              "pause": false,
              "photo": "",
              "start": false,
            },
          },
        },
      }
    }
    history={
      Object {
        "push": [MockFunction],
      }
    }
    key="0-000"
    local={
      Object {
        "rsvp": Object {
          "date": 0,
          "hidden": Array [],
          "id": "",
          "message": "",
          "time": "",
          "user": Object {
            "id": "",
            "name": "",
          },
          "venue": Object {
            "id": "",
            "name": "",
          },
        },
      }
    }
    location={Object {}}
    match={Object {}}
    staticContext={Object {}}
  />
</MetaList>
`;
